packages/vscode-integration/src/types.ts

```diff
@@ -0,0 +1,196 @@
+     * to the webview.
+     *
+     * Use the properties `onBeforeReceiveMessageFromServer` and `onBeforePropagateMessageToClient`
+     * of the GlspVscodeAdapter in order to control what messages are propagated
+     * and processed.
+     */
+    readonly onClientReceiveEmitter: vscode.EventEmitter<unknown>;
+
+    /**
+     * This event is used to notify the VSCode integration about messages from the
+     * client.
+     *
+     * Fire this event with the message the client wants to send to the server.
+     *
+     * Use the properties `onBeforeReceiveMessageFromClient` and `onBeforePropagateMessageToServer`
+     * of the GlspVscodeAdapter in order to control what messages are propagated
+     * and processed.
+     */
+    readonly onClientSend: vscode.Event<unknown>;
+}
+
+/**
+ * The server or server wrapper used by the VSCode integration needs to implement
+ * this interface.
+ */
+export interface GlspVscodeServer {
+
+    /**
+     * An event emitter used by the VSCode extension to send messages to the server.
+     * You should listen to the event attached to this emitter to receive messages
+     * from the client/VSCode integration and send it to the server.
+     *
+     * Use the properties `onBeforeReceiveMessageFromClient` and `onBeforePropagateMessageToServer`
+     * of the GlspVscodeAdapter in order to control what messages are propagated
+     * and processed.
+     */
+    readonly onServerReceiveEmitter: vscode.EventEmitter<unknown>;
+
+    /**
+     * An event the VSCode integration uses to receive messages from the server.
+     * The messages are then propagated to the client or processed by the VSCode
+     * integration to provide functionality.
+     *
+     * Fire this event with the message you want to send to the client.
+     *
+     * Use the properties `onBeforeReceiveMessageFromServer` and `onBeforePropagateMessageToClient`
+     * of the GlspVscodeAdapter in order to control what messages are propagated
+     * and processed.
+     */
+    readonly onServerSend: vscode.Event<unknown>;
+}
+
+interface InterceptorCallback {
+    /**
+     * This callback controls what message should be propagated to the VSCode integration
+     * and whether the VSCode integration should process it (ie. provide functionality
+     * based on the message).
+     *
+     * @param newMessage The message to be propagated. This value can be anything,
+     * however if it is `undefined` the message will not be propagated further.
+     * @param shouldBeProcessedByAdapter Optional parameter indicating whether the
+     * VSCode integration should process the message. That usually means providing
+     * functionality based on the message but also modifying it or blocking it from
+     * being propagated further.
+     */
+    (newMessage: unknown | undefined, shouldBeProcessedByAdapter?: boolean): void;
+}
+
+export interface GlspVscodeAdapterConfiguration {
```

Comment:
Should probably be renamed to `GlspVscodeAdapterOptions` to be consistent with the  naming of other constructor options

Reply:
Renamed in 26e430709fab8d764971bf477ba694e2595aa6a2